### PR TITLE
Project 2

### PR DESCRIPTION
**Pull request by Reggeditt**
- Created a class Nameable.
      > Implement a method called correct_name that will raise a NotImplementedError.
- Made sure that Person class inherits from Nameable
- Prepared base Decorator
- Made sure that it inherits from Nameable.
- Prepared CapitalizeDecorator and TrimmerDecorator

### CODE REVIEW

**Comment by julie-ify** [comment]
Hi @Reggeditt,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

* Kindly remove the live demo section link on the readme header since you have no live demo section associated with it.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: person.rb [hunk 1]
```diff
@@ -1,4 +1,5 @@
-class Person
+require './Nameable'
+class Person < Nameable
```
Comment (VICTOR-CHUKWU) [inline person.rb:2]: - [ ] I can not see any method `.correct_name` in this person class. kindly add it

File: decorator.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      @nameable.correct_name.capitalize
+    end
+end
+  
+class TrimmerDecorator < Decorator
+    def correct_name
+      name = @nameable.correct_name
+      name.length > 10 ? name[0, 10] : name
+    end
+end
```
Comment (VICTOR-CHUKWU) [inline decorator.rb]: - [ ] [Optional] It would be nice if these classes are put in their individual file for easy readability of the codebase

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @Reggeditt  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (Reggeditt) [reply person.rb:2]: fixed

File: decorator.rb (hunk 2 above)
Comment (Reggeditt) [reply decorator.rb]: yes please

Comment (Reggeditt) [reply decorator.rb]: fixed

File: decorator.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require './nameable'
+require './person'
+class Decorator < Nameable
+    def initialize(nameable)
+        @nameable = Nameable
```
Comment (julie-ify) [inline decorator.rb]: - [ ] You should use the small letter `nameable`, instead of `Nameable` as it is causing errors.

  ```
  @nameable = nameable
  ```

**Review by julie-ify — CHANGES_REQUESTED** [review]
Hi @Reggeditt,

## Highlights!
- Good code organization and working app ✔️ 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: decorator.rb (hunk 3 above)
Comment (Reggeditt) [reply decorator.rb]: fixed

**Review by DeliceLydia — APPROVED** [review]
## Hi @Reggeditt, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 

## Highlights
- Addressed changes requested



### Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**